lib/client.js

```diff
@@ -1538,7 +1548,9 @@ function write (client, request) {
     contentLength = null
   }
 
-  if (request.contentLength !== null && request.contentLength !== contentLength) {
+  // https://github.com/nodejs/undici/issues/2046
+  // A user agent may send a Content-Length header with 0 value, this should be allowed.
+  if (shouldSendContentLength(contentLength, method) && request.contentLength !== null && request.contentLength !== contentLength) {
```

Comment:
Following previous comment, it is better if we offload the `contentLength` to the `if` statement instead. In that sense **if** `contentLength` is greater than `0`, and the `method` has explicit semantics to handle the request's body; we attach it. Otherwise, we just ignore it.